codec/reflectcodec/type_codec.go

```diff
@@ -332,6 +352,67 @@ func (c *genericCodec) marshal(value reflect.Value, p *wrappers.Packer, maxSlice
 				return err
 			}
 		}
+		return nil
+	case reflect.Map:
+		keys := value.MapKeys()
+		numElts := len(keys)
+		if uint32(numElts) > maxSliceLen {
+			return fmt.Errorf("%w; map length, %d, exceeds maximum length, %d",
+				codec.ErrMaxSliceLenExceeded,
+				numElts,
+				maxSliceLen,
+			)
+		}
+		p.PackInt(uint32(numElts)) // pack # elements
+		if p.Err != nil {
+			return p.Err
+		}
+
+		// pack key-value pairs sorted by increasing key
+		type keyTuple struct {
+			key   reflect.Value
+			bytes []byte
+		}
+
+		keyPacker := wrappers.Packer{
+			MaxSize: int(c.maxSliceLen),
+			Bytes:   make([]byte, 0, 128),
+		}
+
+		sortedKeys := make([]keyTuple, len(keys))
+		lastKeyPosition := 0
+		for i, key := range keys {
+			// Note this [p] shadows the outer [p]
+			if err := c.marshal(key, &keyPacker, c.maxSliceLen); err != nil {
+				return err
+			}
+			if p.Err != nil {
+				return fmt.Errorf("couldn't marshal map key %+v: %w ", key, p.Err)
+			}
```

Comment:
should this be removed?

Reply:
Good catch @darioush, I addressed it in 684f0b49b